base/applications/cleanmgr/cleanmgr/cleanmgr.cpp

```diff
@@ -271,16 +300,19 @@ class CCleanMgrModule : public ATL::CAtlExeModuleT< CCleanMgrModule >
         psh.dwFlags = PSH_NOAPPLYNOW | PSH_USEICONID | PSH_NOCONTEXTHELP;
         psh.hInstance = _AtlBaseModule.GetResourceInstance();
         psh.pszIcon = MAKEINTRESOURCEW(IDI_CLEANMGR);
-        CStringW Title;
-        Title.Format(IDS_PROPERTIES_MAIN_TITLE, m_Drive);
         psh.pszCaption = Title;
         psh.nPages = _countof(hpsp);
         psh.phpage = hpsp;
 
         if (PropertySheetW(&psh) >= 1)
         {
+            ::DestroyWindow(hWndInstance); // Allow new "cleanmgr /D" without waiting for these handlers
             Handlers.ExecuteCleanup(&CacheCallBack);
         }
+        else
+        {
+            ::DestroyWindow(hWndInstance);
+        }
```

Comment:
```suggestion
        ::DestroyWindow(hWndInstance); // Allow new "cleanmgr /D" without waiting for these handlers
        if (PropertySheetW(&psh) >= 1)
            Handlers.ExecuteCleanup(&CacheCallBack);
```
Wouldn't this work better?

Reply:
Because PropertySheetW is a modal dialog that I need to wait for.

Reply:
Thank you for explanation!